Extension/src/LanguageServer/extension.ts

```diff
@@ -1413,13 +1413,28 @@ export async function preReleaseCheck(): Promise<void> {
     }
 }
 
-export async function getIncludes(maxDepth: number): Promise<any> {
-    const tokenSource = new vscode.CancellationTokenSource();
-    const includes = await clients.ActiveClient.getIncludes(maxDepth, tokenSource.token);
-    tokenSource.dispose();
+export async function getIncludesWithCancellation(maxDepth: number, token: vscode.CancellationToken): Promise<GetIncludesResult> {
+    const includes = await clients.ActiveClient.getIncludes(maxDepth, token);
```

Comment:
Why are you not doing this filtering at the source in the cpptools-srv process? Also, it looks like includes from other workspace folders might be missing at that location.

Reply:
The cpptools-srv side should know if a folder is a system include or not (although there's a bug in that I have a PR to fix).

Reply:
Maybe due to maxDepth being 1.

Reply:
Oh, never mind, I was looking at the wrong data in the debugger. I see includes from other workspace folders now.

Reply:
@sean-mcmanus I agree that trying excluding only the system headers is what we should experiment with in the future. As of now, we only want to restore the functionality exactly as it was.